Names: Ricardo Leão Cardoso, Alexandrine Dif, Annabelle Ducret

# Experiment 1

In [1]:
import numpy as np
from numpy import genfromtxt

# Load signal
data = genfromtxt('ECG.csv', delimiter=',')

# Perform SVD
U1, S1, V1 = np.linalg.svd(data[:, 0:6], full_matrices=False)
print(S1)

# Perform SVD
U2, S2, V2 = np.linalg.svd(data[:, 6:12], full_matrices=False)
print(S2)


[4.79818570e-01 3.12183558e-01 6.64234170e-08 6.26873541e-08
 4.87422030e-08 3.92846440e-08]
[1.23279959 0.76958759 0.35679615 0.21497559 0.17087625 0.10968347]


In [2]:
# Add here the code to compute the effective rank
np.cumsum(S1*S1)/np.sum(S1*S1)


array([0.70258406, 1.        , 1.        , 1.        , 1.        ,
       1.        ])

In [3]:
# Add here the code to compute the effective rank
np.cumsum(S2**2)/np.sum(S2**2)

array([0.65316754, 0.90770737, 0.96241903, 0.9822808 , 0.99482963,
       1.        ])

a) Explain the difference between singular values of the six first columns, and of the last 6 columns.  

The fist thing that one can note is the that the singular value of the S1 are smaller than those in S2. One of the reasons for this could be tha the leads in the first 6 columns are correlated bewteen each other since the combinations between the hand and legs might be redonnant, whereas the last 6 lead are closer to the heart. These collect more precise signal of the electrical field of the heart.

b) Computate of the effective rank with a threshold of 0.98.

From the variance calculation above, we can deduce that the S1 effective rank with a treshold of 0.98 would be 2 and for S2, it would be rank 4.